contracts/base/ModuleManager.sol

```diff
@@ -51,13 +91,13 @@ abstract contract ModuleManager is SelfAuthorized, Executor, GuardManager, IModu
         bytes memory data,
         Enum.Operation operation
     ) internal returns (address guard, bytes32 guardHash) {
-        guard = getGuard();
+        guard = getModuleGuard();
 
         // Only whitelisted modules are allowed.
         require(msg.sender != SENTINEL_MODULES && modules[msg.sender] != address(0), "GS104");
 
         if (guard != address(0)) {
-            guardHash = Guard(guard).checkModuleTransaction(to, value, data, operation, msg.sender);
+            guardHash = IModuleGuard(guard).checkTransaction(to, value, data, operation, msg.sender);
```

Comment:
In order to support a guard that `is IGuard, IModuleGuard`, I would actually keep the `checkModuleTransaction` name.

Reply:
https://github.com/safe-global/safe-smart-account/pull/758/commits/3e4f790849556a5bcc815beb3936af16a3e45ce2